Add emails for pledge receipt and admin notification

rafecolton opened the pull request:
@ericnicolaas @helgatheviking thank you for your patience. This one closes https://github.com/Charitable/Charitable/issues/33

ericnicolaas commented on charitable.php:
```diff
@@ -240,6 +240,8 @@ private function load_dependencies() {
             require_once( $includes_path . 'emails/class-charitable-email-donation-receipt.php' );
             require_once( $includes_path . 'emails/class-charitable-email-campaign-end.php' );
             require_once( $includes_path . 'emails/class-charitable-email-password-reset.php' );
+            require_once( $includes_path . 'emails/class-charitable-email-pledge-notification.php' );
```
Can we change this to `class-charitable-email-offline-donation-notification.php`?

ericnicolaas commented on charitable.php:
```diff
@@ -240,6 +240,8 @@ private function load_dependencies() {
             require_once( $includes_path . 'emails/class-charitable-email-campaign-end.php' );
             require_once( $includes_path . 'emails/class-charitable-email-password-reset.php' );
+            require_once( $includes_path . 'emails/class-charitable-email-pledge-notification.php' );
+            require_once( $includes_path . 'emails/class-charitable-email-pledge-receipt.php' );
```
As above, change to `class-charitable-email-offline-donation-receipt.php`

ericnicolaas commented on includes/emails/charitable-email-hooks.php:
```diff
@@ -28,9 +28,13 @@
  *
  * @see     Charitable_Email_Donation_Receipt::send_with_donation_id()
  * @see     Charitable_Email_New_Donation::send_with_donation_id()
+ * @see     Charitable_Email_Pledge_Receipt::send_with_donation_id()
```
`Charitable_Email_Offline_Donation_Receipt::send_with_donation_id()`

ericnicolaas commented on includes/emails/charitable-email-hooks.php:
```diff
@@ -28,9 +28,13 @@
  * @see     Charitable_Email_Donation_Receipt::send_with_donation_id()
  * @see     Charitable_Email_New_Donation::send_with_donation_id()
+ * @see     Charitable_Email_Pledge_Receipt::send_with_donation_id()
+ * @see     Charitable_Email_Pledge_Notification::send_with_donation_id()
```
`Charitable_Email_Offline_Donation_Notification::send_with_donation_id()`

ericnicolaas commented on includes/emails/charitable-email-hooks.php:
```diff
@@ -28,9 +28,13 @@
 add_action( 'charitable_after_save_donation', array( 'Charitable_Email_Donation_Receipt', 'send_with_donation_id' ) );
 add_action( 'charitable_after_save_donation', array( 'Charitable_Email_New_Donation', 'send_with_donation_id' ) );
+add_action( 'charitable_after_save_donation', array( 'Charitable_Email_Pledge_Receipt', 'send_with_donation_id' ) );
+add_action( 'charitable_after_save_donation', array( 'Charitable_Email_Pledge_Notification', 'send_with_donation_id' ) );
```
Replace `Charitable_Email_Pledge_Notification` with `Charitable_Email_Offline_Donation_Notification`.

ericnicolaas commented on includes/emails/charitable-email-hooks.php:
```diff
@@ -28,9 +28,13 @@
  */
 add_action( 'charitable_after_save_donation', array( 'Charitable_Email_Donation_Receipt', 'send_with_donation_id' ) );
 add_action( 'charitable_after_save_donation', array( 'Charitable_Email_New_Donation', 'send_with_donation_id' ) );
+add_action( 'charitable_after_save_donation', array( 'Charitable_Email_Pledge_Receipt', 'send_with_donation_id' ) );
```
Replace `Charitable_Email_Pledge_Receipt` with `Charitable_Email_Offline_Donation_Receipt`.

ericnicolaas commented on includes/emails/class-charitable-email-pledge-notification.php:
```diff
@@ -0,0 +1,102 @@
+defined( 'ABSPATH' ) or die( 'NO, HUMAN' );
+
+if ( ! class_exists( 'Charitable_Email_Pledge_Notification' ) && class_exists( 'Charitable_Email_New_Donation' ) ) {
+  class Charitable_Email_Pledge_Notification extends Charitable_Email_New_Donation {
```
Rename class to `Charitable_Email_Offline_Donation_Notification`.

ericnicolaas commented on includes/emails/class-charitable-email-pledge-notification.php:
```diff
@@ -0,0 +1,102 @@
+    public function __construct( $objects = array() ) {
+      parent::__construct( $objects );
+
+      $this->name = apply_filters( 'charitable_email_pledge_notification_name', __( 'New Pledge Notification', 'charitable' ) );
```
Change `charitable_email_pledge_notification_name` to `charitable_email_offline_donation_notification_name` and `New Pledge Notification` to `New Offline Donation Notification`.

ericnicolaas commented on includes/emails/class-charitable-email-pledge-notification.php:
```diff
@@ -0,0 +1,102 @@
+      }
+
+      /* All three of those checks passed, so proceed with sending the email. */
+      $email = new Charitable_Email_Pledge_Notification( array(
```
Change class name.

ericnicolaas commented on includes/emails/class-charitable-email-pledge-receipt.php:
```diff
@@ -0,0 +1,122 @@
+
+defined( 'ABSPATH' ) or die( 'NO, HUMAN' );
+
+if ( ! class_exists( 'Charitable_Email_Pledge_Receipt' ) && class_exists( 'Charitable_Email_Donation_Receipt' ) ) {
```
Change class name to `Charitable_Email_Offline_Donation_Receipt`.

ericnicolaas commented on includes/emails/class-charitable-email-pledge-receipt.php:
```diff
@@ -0,0 +1,122 @@
+
+if ( ! class_exists( 'Charitable_Email_Pledge_Receipt' ) && class_exists( 'Charitable_Email_Donation_Receipt' ) ) {
+
+  class Charitable_Email_Pledge_Receipt extends Charitable_Email_Donation_Receipt {
```
Change class name to `Charitable_Email_Offline_Donation_Receipt`.

ericnicolaas commented on includes/emails/class-charitable-email-pledge-receipt.php:
```diff
@@ -0,0 +1,122 @@
+    /**
+     * @var     string
+     */
+    CONST ID = 'pledge_receipt';
```
Replace with `offline_donation_receipt`.

ericnicolaas commented on includes/emails/class-charitable-email-pledge-notification.php:
```diff
@@ -0,0 +1,102 @@
+    /**
+     * @var     string
+     */
+    CONST ID = 'new_pledge';
```
Change to `offline_donation_notification`.

ericnicolaas commented on includes/emails/class-charitable-email-pledge-receipt.php:
```diff
@@ -0,0 +1,122 @@
+     */
+    public function __construct( $objects = array() ) {
+      parent::__construct( $objects );
+      $this->name = apply_filters( 'charitable_email_pledge_receipt_name', __( 'Pledge Receipt', 'charitable' ) );
```
- [ ] `charitable_email_pledge_receipt_name` becomes `charitable_email_offline_donation_receipt_name`.
- [ ] `Pledge Receipt` becomes `Offline Donation Receipt`

ericnicolaas commented on includes/emails/class-charitable-email-pledge-receipt.php:
```diff
@@ -0,0 +1,122 @@
+     * @return  string
+     * @access  public
+     * @static
+     * @since   1.0.3
```
Since should be 1.5.0.

ericnicolaas commented on includes/emails/class-charitable-email-pledge-receipt.php:
```diff
@@ -0,0 +1,122 @@
+     * @return  boolean
+     * @access  public
+     * @static
+     * @since   1.0.0
```
Since should be 1.5.0.

ericnicolaas commented on includes/emails/class-charitable-email-pledge-receipt.php:
```diff
@@ -0,0 +1,122 @@
+     *
+     * @return  string
+     * @access  protected
+     * @since   1.0.0
```
Since should be 1.5.0.

ericnicolaas commented on includes/emails/class-charitable-email-pledge-receipt.php:
```diff
@@ -0,0 +1,122 @@
+     *
+     * @return  string
+     * @access  protected
+     * @since   1.0.0
```
Since should be 1.5.0.

ericnicolaas commented on includes/emails/class-charitable-emails.php:
```diff
@@ -87,6 +87,8 @@ public function register_emails() {
 				'new_donation'     => 'Charitable_Email_New_Donation',
 				'campaign_end'     => 'Charitable_Email_Campaign_End',
 				'password_reset'   => 'Charitable_Email_Password_Reset',
+				'pledge_receipt'   => 'Charitable_Email_Pledge_Receipt',
```
- [ ] `pledge_receipt` becomes `offline_donation_receipt`
- [ ] `Charitable_Email_Pledge_Receipt` becomes `Charitable_Email_Offline_Donation_Receipt`

ericnicolaas commented on includes/emails/class-charitable-emails.php:
```diff
@@ -87,6 +87,8 @@ public function register_emails() {
 				'campaign_end'     => 'Charitable_Email_Campaign_End',
 				'password_reset'   => 'Charitable_Email_Password_Reset',
+				'pledge_receipt'   => 'Charitable_Email_Pledge_Receipt',
+				'new_pledge'       => 'Charitable_Email_Pledge_Notification',
```
- [ ] `new_pledge` becomes `offline_donation_notification`
- [ ] `Charitable_Email_Pledge_Notification` becomes `Charitable_Email_Offline_Donation_Notification`

helgatheviking commented:
I kinda like the "pledge" nomenclature, but can see it may not make sense to change at this point.

ericnicolaas commented:
Yeah, it does work well. I think it's confusing for users though if there
is an email that only is used for offline donations, but the email refers
to pledges instead.

Plus, one potential add-on in future would add support for pre-approved
payments (i.e. Kickstarter-style) and I had thought pledge could work well
in that case.

On Thu., 24 Aug. 2017, 9:19 pm Kathy <notifications@github.com> wrote:

> I kinda like the "pledge" nomenclature, but can see it may not make sense
> to change at this point.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Charitable/Charitable/pull/380#issuecomment-324613157>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AApS5n8xQBPE2ZB9VxswU0YAdPSXGyaRks5sbWNUgaJpZM4O9ada>
> .
>
-- 


*Eric Daams*Co-creator of Charitable

wpcharitable.com
twitter.com/wpcharitable

The fundraising alternative for non-profits.
Powered by WordPress.

rafecolton commented:
I suppose my nomenclature has revealed my use case for offline donations 😄 

@ericnicolaas I'll make the changes ASAP

rafecolton commented:
@ericnicolaas I believe I have addressed all of your comments - nomenclature and since tags. Please let me know if you would like me to make any more changes. Thank you for reviewing!

ericnicolaas commented:
Merged! Thanks @rafecolton!
